src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBinaryWriter.cs

```diff
@@ -17,13 +17,13 @@ namespace System.Xml
 {
     public interface IXmlBinaryWriterInitializer
     {
-        void SetOutput(Stream stream, IXmlDictionary dictionary, XmlBinaryWriterSession session, bool ownsStream);
+        void SetOutput(Stream stream, IXmlDictionary? dictionary, XmlBinaryWriterSession? session, bool ownsStream);
```

Comment:
w.r.t. `IXmlBinaryReaderInitializer.SetInput` [comment ](https://github.com/dotnet/runtime/pull/41476/files#r479457449): nullable dictionary and session are handled without throwing in this case.

Reply:
The same reasoning exists here, we have public overloads to `CreateBinaryWriter` that explicitly pass `null` to these parameters.

https://github.com/dotnet/runtime/blob/d5d046d03589dedc4288515ddde90b3e804c3dc2/src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlDictionaryWriter.cs#L20-L33